### PR TITLE
Add "related programs" section to "application confirmation" page

### PR DESCRIPTION
**Pull request by ssandbekkhaug**
### Description

Add "related programs" section to "application confirmation" page. The cards are not fully styled yet.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Issue(s) this completes

Part of #8306

### CODE REVIEW

File: server/app/views/applicant/ApplicantUpsellTemplate.html [hunk 1]
```diff
@@ -127,11 +127,24 @@ <h2 th:text="#{heading.nextSteps}"></h2>
...
-            <!-- TODO(#8306): Add related programs section -->
           </div>
         </div>
       </div>
+
+      <section th:if="${section.cards().size() > 0}">
+        <div class="grid-container section">
+          <div class="grid-row">
+            <div class="tablet:grid-col-8">
+              <div class="usa-prose">
```
Comment (melanie-exygy) [inline server/app/views/applicant/ApplicantUpsellTemplate.html:138]: **praise:** Neat that you're already using this!

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss [hunk 2]
```diff
@@ -212,13 +212,14 @@ main {
...
     grid-template-columns: 1fr 1fr;
+    margin-left: -0.5rem;
+    margin-right: -0.5rem;
   }
 }
 
 // Desktop: Program card grid has 3 columns
 @media (min-width: 64em) {
   .cf-program-card-group {
-    display: grid;
```
Comment (melanie-exygy) [inline server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:221]: **question (non-blocking):** Why remove grid on desktop?

File: browser-test/src/applicant/northstar_upsell.test.ts [hunk 3]
```diff
@@ -62,9 +74,10 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
 
     await validateAccessibility(page)
 
-    await test.step('Validate that user can click through without logging in', async () => {
+    await test.step('Validate that user can return to home page without logging in', async () => {
       await applicantQuestions.clickBackToHomepageButton()
-      await expect(page.locator('[data-testId="login"]')).toBeHidden()
+      // Expect the login dialog did not appear, so the user should already see the homepage
+      await expect(page.locator('[data-testId="home"]')).toBeVisible()
```
Comment (melanie-exygy) [inline browser-test/src/applicant/northstar_upsell.test.ts:80]: **suggestion (non-blocking):** Could we ensure that they are still logged in by checking something like this?
```
   await expect(page.getByRole('banner')).toContainText(
      `Logged in as`,
    )

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss (hunk 2 above)
Comment (ssandbekkhaug) [reply server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:221]: I realized `grid` in the `min-width:40em` rule also applies to desktop, so it's redundant.

Comment (melanie-exygy) [reply server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:221]: good catch!

File: server/app/controllers/applicant/UpsellController.java [hunk 4]
```diff
@@ -137,38 +137,46 @@ public CompletionStage<Result> considerRegister(
...
-                      // we are already checking if profile is empty
+                      // We are already checking if profile is empty
                       v ->
                           applicantService.maybeEligibleProgramsForApplicant(
                               applicantId, profile.get(), request),
                       classLoaderExecutionContext.current())
-                  .thenApplyAsync(Optional::of);
+                  .thenApplyAsync(
+                      eligiblePrograms -> {
+                        // Don't show the program that the user just finished applying to
```
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/controllers/applicant/UpsellController.java:157]: Thank you!

File: browser-test/src/applicant/northstar_upsell.test.ts [hunk 5]
```diff
@@ -62,9 +74,14 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
 
     await validateAccessibility(page)
 
-    await test.step('Validate that user can click through without logging in', async () => {
+    await test.step('Validate that user can return to the home page without logging in', async () => {
       await applicantQuestions.clickBackToHomepageButton()
-      await expect(page.locator('[data-testId="login"]')).toBeHidden()
+      // Expect the login dialog did not appear, so the user should already see the homepage
+      await expect(page.locator('[data-testId="home"]')).toBeVisible()
```
Comment (gwendolyngoetz) [inline browser-test/src/applicant/northstar_upsell.test.ts:80]: (blocking) There should only ever be a single `body` tag. Have your locator look for that so we don't need to add the `data-testId`.

File: server/app/views/applicant/ProgramIndexTemplate.html [hunk 6]
```diff
@@ -33,7 +33,7 @@
   <head
     th:replace="~{applicant/ApplicantBaseFragment :: pageHeaderScriptsAndLinks}"
   ></head>
-  <body>
+  <body data-testId="home">
```
Comment (gwendolyngoetz) [inline server/app/views/applicant/ProgramIndexTemplate.html:36]: See previous comment

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**